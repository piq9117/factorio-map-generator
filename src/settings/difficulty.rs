use serde::ser::SerializeStruct;
use serde::{Serialize, Serializer};

pub struct DifficultySettings {
    pub technology_price_multiplier: Option<f32>,
    pub spoil_time_modifier: Option<f32>,
}

impl Serialize for DifficultySettings {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        let mut s = serializer.serialize_struct("DifficultySettings", 2)?;
        s.serialize_field(
            "technology_price_multiplier",
            &self.technology_price_multiplier.unwrap_or(0.001),
        )?;
        s.serialize_field(
            "spoil_time_modifier",
            &self.spoil_time_modifier.unwrap_or(0.0),
        )?;
        s.end()
    }
}
