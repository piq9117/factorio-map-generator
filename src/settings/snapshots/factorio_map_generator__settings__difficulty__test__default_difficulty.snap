---
source: src/settings/difficulty.rs
expression: default_difficulty
---
technology_price_multiplier: 0.001
spoil_time_modifier: 0
